docs/reference/analysis/tokenizers/edgengram-tokenizer.asciidoc

```diff
@@ -1,80 +1,190 @@
---------------------------------------------------
+GET _cluster/health?wait_for_status=yellow
 
-[float]
-==== `side` deprecated
+POST my_index/_analyze
+{
+  "analyzer": "my_analyzer",
+  "text": "2 Quick Foxes."
+}
+----------------------------
+// CONSOLE
 
-There used to be a `side` parameter up to `0.90.1` but it is now deprecated. In
-order to emulate the behavior of `"side" : "BACK"` a
-<<analysis-reverse-tokenfilter,`reverse` token filter>>  should be used together
-with the <<analysis-edgengram-tokenfilter,`edgeNGram` token filter>>. The
-`edgeNGram` filter must be enclosed in `reverse` filters like this:
+The above example produces the following terms:
 
-[source,js]
---------------------------------------------------
-    "filter" : ["reverse", "edgeNGram", "reverse"]
---------------------------------------------------
+[source,text]
+---------------------------
+[ Qu, Qui, Quic, Quick, Fo, Fox, Foxe, Foxes ]
+---------------------------
+
+[TIP]
+.Index-time vs Search-time
+===========================================
+
+Usually we recommend using the same `analyzer` at index time and at search
+time. In this case, the advice is different.  It only makes sense to use the
+`edge_ngram` tokenizer at index time, to ensure that partial words are
+available for matching in the index.  At search time, just search for the
+terms the user has typed in, for instance: `Quick Fo`.
+
+Below is an example of how to set up a field for _search-as-you-type_:
 
-which essentially reverses the token, builds front `EdgeNGrams` and reverses
-the ngram again. This has the same effect as the previous `"side" : "BACK"` setting.
+[source,js]
+-----------------------------------
+PUT my_index
+{
+  "settings": {
+    "analysis": {
+      "analyzer": {
+        "autocomplete": {
+          "tokenizer": "autocomplete",
+          "filter": [
+            "lowercase"
+          ]
+        },
+        "autocomplete_search": {
+          "tokenizer": "lowercase"
+        }
+      },
+      "tokenizer": {
+        "autocomplete": {
+          "type": "edge_ngram",
+          "min_gram": 2,
+          "max_gram": 10,
+          "token_chars": [
+            "letter"
+          ]
+        }
+      }
+    }
+  },
+  "mappings": {
+    "doc": {
+      "properties": {
+        "title": {
+          "type": "text",
+          "analyzer": "autocomplete",
+          "search_analyzer": "autocomplete_search"
+        }
+      }
+    }
+  }
+}
+
+PUT my_index/doc/1
+{
+  "title": "Quick Foxes" <1>
+}
+
+GET my_index/_search
+{
+  "query": {
+    "match": {
+      "title": {
+        "query": "Quick Fo", <2>
+        "operator": "and"
+      }
+    }
+  }
+}
+-----------------------------------
+// CONSOLE
+<1> The `autocomplete` analyzer indexes the terms `[qu, qui, quic, quick, fo, fox, foxe, foxes]`.
+<2> The `autocomplete_search` analyzer searches for the terms `[quick, fo]`, both of which appear in the index.
+
+===========================================
 
```

Comment:
It'd be nice to have a `// TESTRESPONSE` snippet here so we can be sure that the example works and so people see what is found. It'd be even nicer to have a document indexed that *won't* be found by the search but might be if the wrong search_analyzer is used.